src/rosdiscover/config.py

```diff
@@ -2,14 +2,68 @@
 __all__ = ('Config',)
 
 from types import MappingProxyType
-from typing import Any, Mapping, Sequence
+import typing as t
 
 from loguru import logger
 import attr
 import roswire
 import yaml
 
 
+@attr.s(frozen=True, slots=True, auto_attribs=True)
+class NodeSourceInfo:
+    """
+    This is a class to represent nodes and their sources, as a bootstrap for recovering node information
+    from static analysis.
+
+    Note: This is an interim class, and will be replaced by CMakelists.txt analysis in the fullness of time.
```

Comment:
Need this be an interim class? Could this not be data structure that's produced by the CMake analysis?

Reply:
We haven't pulled in the ros-anatomy stuff to rosdiscover.